jobs/build/ocp/Jenkinsfile

```diff
@@ -187,285 +242,346 @@ node(TARGET_NODE) {
-            }
+                  oit """
+--working-dir ${OIT_WORKING} --group 'openshift-${BUILD_VERSION}' \\
+distgits:rebase --sources ${env.WORKSPACE}/sources.yml --version ${NEW_VERSION} \\
+--release ${NEW_DOCKERFILE_RELEASE} --no-push
+"""
+                }
 
-            stage( "build images" ) {
-                // TODO: Create a dynamic .repo file pointing to the exact puddle we built instead of "building" so that we can run X.Y builds in parallel
-                sh "ose_images.sh --user ocp-build build_container --branch rhaos-${BUILD_VERSION}-rhel-7 --group base --repo https://raw.githubusercontent.com/openshift/aos-cd-jobs/master/build-scripts/repo-conf/aos-unsigned-building.repo"
-            }
+                record_log = buildlib.parse_record_log( OIT_WORKING )
+                distgit_notify = buildlib.get_distgit_notify( record_log )
+
+                for(kv in buildlib.mapToList(distgit_notify)) {
+                    distgit = kv[0]
+                    val = kv[1]
+                    if(val.owners) {
+                        alias = val['source_alias']
+                        dockerfile_url = ""
+                        if ( alias ){
+                            github_url = env.GITHUB_URLS[alias]
+                            github_url = github_url.minus(".git")
+                            github_url = github_url.minus("git@")
+                            github_url = github_url.replaceFirst(":", "/")
+                            base_path = env.GITHUB_BASE_PATHS[alias]
+                            dockerfile_sub_path = val['dockerfile'].minus(base_path)
+                            dockerfile_url = "Source file: " + github_url + "/" + dockerfile_sub_path
+                        }
+                        mail(to: "${val.owners}",
+                                from: "aos-cd@redhat.com",
+                                subject: "${val.image} Dockerfile reconciliation.",
+                                body: """OIT has detected a change in the Dockerfile for ${val.image}
+                    ${dockerfile_url}
+                    This has been automatically reconciled and the new file can be seen here:
+                    http://pkgs.devel.redhat.com/cgit/${disgit}/tree/Dockerfile?id=${val.sha}
+                    """);
+                    }
+                }
 
-            if ( EARLY_LATEST_HACK.toBoolean() ) {
+                stage( "build images" ) {
+                    sh """
+ose_images.sh --user ocp-build build_container --branch rhaos-${BUILD_VERSION}-rhel-7 --group base \\
+--repo https://raw.githubusercontent.com/openshift/aos-cd-jobs/master/build-scripts/repo-conf/aos-unsigned-building.repo"
+"""
+
+                    oit """
+--working-dir ${OIT_WORKING} --group openshift-${BUILD_VERSION} --include aos3-installation-docker \\
+distgits:build-images \\
+--push-to-defaults --repo_type unsigned --scratch
```

Comment:
Why is this scratch?

Reply:
I asked you about that in a previous comment... I was just being safe for now.

Reply:
--scatch means the image will be temporary and with an unpredicable name/tag. We can't use it for the build process.